File: app/src/main/java/com/example/client/models/profile/ProfileModel.java
package com.example.client.models.profile;

public class ProfileModel {
    private int id;
    private String email;
    private String avatar;
    private String fullname;
    private String birthday;
    private String phone;
    private String address;
    private int isactive;
    private int isadmin;
    private Double wallet;

    public ProfileModel() {
    }

    public ProfileModel(int id, String email, String avatar, String fullname, String birthday, String phone, String address, int isactive, int isadmin, Double wallet) {
        this.id = id;
        this.email = email;
        this.avatar = avatar;
        this.fullname = fullname;
        this.birthday = birthday;
        this.phone = phone;
        this.address = address;
        this.isadmin = isadmin;
        this.wallet = wallet;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getFullname() {
        return fullname;
    }

    public void setFullname(String fullname) {
        this.fullname = fullname;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public int getIsactive() {
        return isactive;
    }

    public void setIsactive(int isactive) {
        this.isactive = isactive;
    }

    public int getIsadmin() {
        return isadmin;
    }

    public void setIsadmin(int isadmin) {
        this.isadmin = isadmin;
    }

    public Double getWallet() {
        return wallet;
    }

    public void setWallet(Double wallet) {
        this.wallet = wallet;
    }
}

File: app/src/main/java/com/example/client/models/profile/ProfileModel.kt
package com.example.client.models.profile

import com.example.client.base.BaseModel

data class ProfileModel(
        val id: Int = -1,
        val email: String = "",
        var avatar: String = "",
        var fullname: String = "",
        var birthday: String = "",
        var phone: String = "",
        var csc_point: Double = 0.0,
        var wallet: Double = 0.0
): BaseModel()

data class ProfileResponse(
        val id: Int?,
        val email: String?,
        var avatar: String?,
        var fullname: String?,
        var birthday: String?,
        var phone: String?,
        var wallet: Double?,
        var csc_point: Double?
) : BaseModel() {
    fun toProfileModel() = ProfileModel(
            id = id ?: -1,
            email = email.orEmpty(),
            avatar = avatar.orEmpty(),
            fullname = fullname.orEmpty(),
            birthday = birthday.orEmpty(),
            phone = phone.orEmpty(),
            csc_point = csc_point ?: 0.0,
            wallet = wallet ?: 0.0
    )
}

data class DataProfileResponse(val access_token: String)

data class ProfileRequest(
        var email: String? = null,
        var avatar: String? = null,
        var fullname: String? = null,
        var birthday: String? = null,
        var phone: String? = null,
        var new_password: String? = null,
        var old_password: String? = null,
)